SoC/gd32vf103/Board/gd32vf103v_eval/Source/gd32vf103v_eval.c

```diff
@@ -1,252 +1,252 @@
+static uint8_t KEY_IRQn[KEYn] = {KEY_A_EXTI_IRQn,
+                                 KEY_B_EXTI_IRQn,
+                                 KEY_C_EXTI_IRQn,
+                                 KEY_D_EXTI_IRQn,
+                                 KEY_CET_EXTI_IRQn};
+
+/*!
+    \brief      configure led GPIO
+    \param[in]  lednum: specify the led to be configured
+      \arg        LED1
+      \arg        LED2
+      \arg        LED3
+      \arg        LED4
+    \param[out] none
+    \retval     none
+*/
+void gd_led_init(led_typedef_enum lednum)
+{
+    /* enable the led clock */
+    rcu_periph_clock_enable(GPIO_CLK[lednum]);
+    /* configure led GPIO port */
+    gpio_init(GPIO_PORT[lednum], GPIO_MODE_OUT_PP, GPIO_OSPEED_50MHZ, GPIO_PIN[lednum]);
+
+    GPIO_BC(GPIO_PORT[lednum]) = GPIO_PIN[lednum];
+}
+
+/*!
+    \brief      turn on selected led
+    \param[in]  lednum: specify the led to be turned on
+      \arg        LED1
+      \arg        LED2
+      \arg        LED3
+      \arg        LED4
+    \param[out] none
+    \retval     none
+*/
+void gd_led_on(led_typedef_enum lednum)
+{
+    GPIO_BOP(GPIO_PORT[lednum]) = GPIO_PIN[lednum];
+}
+
+/*!
+    \brief      turn off selected led
+    \param[in]  lednum: specify the led to be turned off
+      \arg        LED1
+      \arg        LED2
+      \arg        LED3
+      \arg        LED4
+    \param[out] none
+    \retval     none
+*/
+void gd_led_off(led_typedef_enum lednum)
+{
+    GPIO_BC(GPIO_PORT[lednum]) = GPIO_PIN[lednum];
+}
+
+/*!
+    \brief      toggle selected led
+    \param[in]  lednum: specify the led to be toggled
+      \arg        LED1
+      \arg        LED2
+      \arg        LED3
+      \arg        LED4
+    \param[out] none
+    \retval     none
+*/
+void gd_led_toggle(led_typedef_enum lednum)
+{
+    gpio_bit_write(GPIO_PORT[lednum], GPIO_PIN[lednum],
+        (bit_status)(1-gpio_input_bit_get(GPIO_PORT[lednum], GPIO_PIN[lednum])));
+}
+
+/*!
+    \brief      configure key
+    \param[in]  key_num: specify the key to be configured
+      \arg        KEY_A: wakeup key
+      \arg        KEY_B: tamper key
+      \arg        KEY_C: user key
+      \arg        KEY_D: user key
+      \arg        KEY_CET: user key
+    \param[in]  key_mode: specify button mode
+      \arg        KEY_MODE_GPIO: key will be used as simple IO
+      \arg        KEY_MODE_EXTI: key will be connected to EXTI line with interrupt
+    \param[out] none
+    \retval     none
+*/
+
+void gd_key_init(key_typedef_enum keynum, keymode_typedef_enum key_mode)
+{
+    ECLIC_ClearPendingIRQ(KEY_IRQn[keynum]);
+    /* enable the key clock */
+    rcu_periph_clock_enable(KEY_CLK[keynum]);
+    rcu_periph_clock_enable(RCU_AF);
+
+    /* configure button pin as input */
+    gpio_init(KEY_PORT[keynum], GPIO_MODE_IN_FLOATING, GPIO_OSPEED_50MHZ, KEY_PIN[keynum]);
+
+    if (key_mode == KEY_MODE_EXTI) {
+        /* enable and set key EXTI interrupt to the lowest priority */
+	ECLIC_EnableIRQ(KEY_IRQn[keynum]);
```

Comment:
This code is not aligned, could you help to align it, using 4 spaces.

Reply:
grrr. I don't know, why this always happens. I've extra running your editor config in notepad++.